### ISSUE TITLE
Error details missing from FirebaseError auth/network-request-failed

### ISSUE THREAD

**Issue by juni0r** [firebase/firebase-js-sdk#6074]
* Operating System version: macOS 12.2.1
  * Browser version: Chrome Version 99.0.4844.51
  * Firebase SDK version: 9.6.8
  * Firebase Product: auth

#### Steps to reproduce:

Call `signInWithEmailAndPassword(auth, email, password)` with an invalid password.

#### Relevant Code:

```javascript
try {
  await signInWithEmailAndPassword(auth, email, password)
} catch (error) {
  error.code // 'auth/network-request-failed'
  error.message // 'Firebase: Error (auth/network-request-failed).'
  error.name // 'FirebaseError'
  error.customData // {}
}
```

The response body of the network request contains additional details that aren't exposed:

```js
{
  "error": {
    "code": 400,
    "message": "INVALID_PASSWORD",
    "errors": [
      {
        "message": "INVALID_PASSWORD",
        "reason": "invalid",
        "domain": "global"
      }
    ]
  }
}
```

**Comment by sam-gc** [comment]
Hi @juni0r, I've tried replicating this but the auth error contains the expected code (auth/wrong-password).

```json
{
  "error": {
    "code": 400,
    "message": "INVALID_PASSWORD",
    "errors": [
      {
        "message": "INVALID_PASSWORD",
        "domain": "global",
        "reason": "invalid"
      }
    ]
  }
}
```
yields
```json
{
    "code": "auth/wrong-password",
    "customData": {},
    "name": "FirebaseError"
}
```

(9.6.8, Chrome/Mac)

Can you share a bit more details? Are you seeing this on more than one machine? How do you build your app?

**Comment by juni0r** [comment]
I'm seeing this on my developent machine using the firebase auth emulator.  It's a Nuxt.js (Vue) app built with webpack.

These are the relvant code portions:

```ts
// lib/firebase.ts

import config from './config'

import { initializeApp } from 'firebase/app'

import type { Auth } from 'firebase/auth'
import {
  initializeAuth,
  connectAuthEmulator,
  browserLocalPersistence as persistence,
  browserPopupRedirectResolver as redirectResolver
} from 'firebase/auth'

const isDev = process.env.NODE_ENV === 'development'

/* APP */
export const firebase = initializeApp(config)

/* AUTH */
let auth: Auth

export function getAuth(): Auth {
  if (!auth) {
    auth = initializeAuth(firebase, { persistence, redirectResolver })
    if (isDev) {
      connectAuthEmulator(auth, `http://localhost:9099`)
    }
  }
  return auth
}
```
```ts
// lib/auth.ts

import { getAuth } from './lib/firebase'
import { signInWithEmailAndPassword } from 'firebase/auth'

export function signIn(email: string, password: string) {
  return signInWithEmailAndPassword(getAuth(), email, password)
}
```

The actual call to the function happens in an event handler, which is essentially

```ts
import { signIn } from '@lib/auth'

async onSubmit() {
  const { email, password } = this
  try {
    await signIn(email, password)
  } catch(error) {
    console.error(error)
   // Uncaught (in promise) FirebaseError: Firebase: Error (auth/network-request-failed).
  }
}
```
The corresponding network request however, comes back with a detailed response as described above, containing `message: "INVALID_PASSWORD"`

**Comment by sam-gc** [comment]
I just tried the basic Nuxt app (I followed https://nuxtjs.org/docs/get-started/installation). What defaults are most applicable to your use case? For reference, this is what I entered:
```
? Project name: nuxt
? Programming language: JavaScript
? Package manager: Yarn
? UI framework: None
? Nuxt.js modules: (Press <space> to select, <a> to toggle all, <i> to invert s
election) - none -
? Linting tools: (Press <space> to select, <a> to toggle all, <i> to invert sel
ection) - none -
? Testing framework: None
? Rendering mode: Universal (SSR / SSG)
? Deployment target: Static (Static/Jamstack hosting)
? Development tools: (Press <space> to select, <a> to toggle all, <i> to invert
 selection) - none -
? What is your GitHub username? 
? Version control system: None
```

With this setup and the code you listed (and testing against both prod and emulator), I'm still seeing the expected error message (`auth/wrong-password`).

**Comment by juni0r** [comment]
@sam-gc  Upon further investigation I realized that we had a Nuxt module in our code that would also import `lib/auth.ts`  and call `getAuth()`. Since Nuxt apps employ SSR, modules will run both in Node and in the browser.

Apparently firebase/auth got initialized in the node environment which caused the issue. I realized this when I tried to create a blank-slate Nuxt app just like you did.

My apologies for raising false alarm and thank you for looking into this. Have a great weekend :)
